File: backend/src/main/java/com/synaxis/backend/room/dto/LeaveRoomRequest.java
package com.synaxis.backend.room.dto;

import jakarta.validation.constraints.NotBlank;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Getter
@Setter
@NoArgsConstructor
public class LeaveRoomRequest {

    @NotBlank(message = "must not be blank")
    private String playerId;
    @NotBlank(message = "must not be blank")
    private String playerToken;
}

File: frontend/lib/features/room/data/models/leave_room_request.dart
class LeaveRoomRequest {
  const LeaveRoomRequest({
    required this.playerId,
    required this.playerToken,
  });

  final String playerId;
  final String playerToken;

  Map<String, dynamic> toJson() => {
        'playerId': playerId,
        'playerToken': playerToken,
      };
}
